crates/proof-of-sql/src/base/database/union_util.rs

```diff
@@ -174,6 +174,27 @@ pub fn column_union<'a, S: Scalar>(
                 }) as &[_],
             )
         }
+        ColumnType::FixedSizeBinary(width) => {
+            let bw = width.width_as_usize();
+            let total_bytes = len * bw;
+            let flattened: Vec<u8> = columns
+                .iter()
+                .flat_map(|col| {
+                    let (col_width, col_data) = col
+                        .as_fixed_size_binary()
+                        .expect("column_type check above should guarantee same type");
+                    assert_eq!(col_width, width, "Inconsistent FixedSizeBinary width");
+                    col_data.iter().copied()
+                })
+                .collect();
```

Comment:
Please don't create temporary `Vec`s. See the other match arms here for examples of how to avoid this.